src/analysis/event_detector.py

```diff
@@ -0,0 +1,312 @@
+import xarray as xr
+import numpy as np
+from pathlib import Path
+from xclim.run_length import rle, longest_run  # , windowed_run_events
+from typing import Tuple, Optional, Any
+import warnings
+import pandas as pd
+from scripts.eng_utils import get_ds_mask
+
+
+class EventDetector():
+    def __init__(self,
+                 path_to_data: Path,
+                 data_folder: Path = Path('data')) -> None:
```

Comment:
Is the `data_folder` necessary? Can it just be inferred from the `path_to_data` (since the chirps file had to be created by a preprocessor)

Reply:
Is it used anywhere? `self.processed_folder` is only called in the `__init__`

Reply:
not necessary for sure - so removed it. thanks!